apps/balancer-tools/src/app/apr/round/(components)/MoreFiltersButton.tsx

```diff
@@ -0,0 +1,266 @@
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+
+  const filters = [
+    {
+      name: "network",
+      label: "Network",
+      options: Object.values(networksOnBalancer).map(capitalize),
+    },
+    { name: "types", label: "Pool type", options: Object.values(PoolTypeEnum) },
+  ];
+
+  const [selectedAttributes, setSelectedAttributes] =
+    useState<SelectedAttributesType>({
+      network: [] as string[],
+      types: [] as string[],
+      minTvl: String(INITIAL_MIN_TVL),
+      maxTvl: null,
+      minApr: null,
+      maxApr: null,
+      minVotingShare: null,
+      maxVotingShare: null,
+    });
+
+  useEffect(() => {
+    const valuesAttributes = [
+      "minTvl",
+      "maxTvl",
+      "minApr",
+      "maxApr",
+      "minVotingShare",
+      "maxVotingShare",
+    ];
+    const arrayAttributes = ["network", "types"];
+    const updatedAttributes = { ...selectedAttributes };
+    valuesAttributes.forEach((value) => {
+      const paramValue = searchParams.get(value);
+      if (paramValue !== null) {
+        updatedAttributes[value] = paramValue;
+      }
+    });
+
+    arrayAttributes.forEach((value) => {
+      if (searchParams.has(value)) {
+        updatedAttributes[value] = searchParams.get(value)?.split(",") || [];
+      }
+    });
+    setSelectedAttributes(updatedAttributes);
+  }, []);
+
+  const pushQueryparamsURL = () => {
+    const current = new URLSearchParams(Array.from(searchParams.entries()));
+
+    Object.entries(selectedAttributes).map(([name, values]) => {
+      if (values && values.length) {
+        current.set(name, Array.isArray(values) ? values.join(",") : values);
+      } else {
+        current.delete(name);
+      }
+    });
+
+    const search = current.toString();
+    const query = search ? `?${search}` : "";
+    router.push(pathname + query, { scroll: false });
+  };
+
+  const handleAttributeChange = (type: string, value: string) => {
+    setSelectedAttributes((prevAttributes) => {
+      const currentAttributes = prevAttributes[type];
+      let updatedAttributes;
+
+      if (Array.isArray(currentAttributes)) {
+        updatedAttributes = currentAttributes.includes(value)
+          ? currentAttributes.filter((v) => v !== value)
+          : [...currentAttributes, value];
+      } else {
+        updatedAttributes = [value];
+      }
+
+      return {
+        ...prevAttributes,
+        [type]: updatedAttributes,
+      };
+    });
+  };
+
+  const handleOnMinMaxChange = (
+    inputName: keyof SelectedAttributesType,
+    value: string,
+  ) => {
+    setSelectedAttributes((prevAttributes) => {
+      const updatedAttributes = {
+        ...prevAttributes,
+        [inputName]: value,
+      };
+      return updatedAttributes;
+    });
+  };
+
+  useEffect(() => {
+    const timer = setTimeout(pushQueryparamsURL, 700);
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [selectedAttributes]);
+
```

Comment:
cool! You could also use the `useDebounce` hook we've defined

Reply:
I tried it and it does'nt work in for this use case, react throws an exception `Error: Invalid hook call. Hooks can only be called inside of the body of a function component.`, because of its useState